### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by juwara0**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [X] #major# - incompatible API change

Closes: https://agile-jira.ciena.com/browse/FROST-1284 and https://agile-jira.ciena.com/browse/FROST-1323

# CHANGELOG
* **Updated** pull request template
* **Added** issue template
* **Updated** to `pr-bumper` version `3`
* **Updated** to node 8
* **Added** slack integration
* **Updated** `ember-frost-test` to `^4.0.1`
* **Updated** `ember-test-utils` to `^8.1.1`
* **Updated** `ember-cli-frost-blueprints` to `^5.0.2`
* **Updated** `ember-prop-types` to `^7.0.1`
* **Updated** `ember-spread` to `^5.0.0`
* **Updated** helper `array.js` to use `array.slice()`
* **Removed** helper `ehook.js` app export and tests since it is now provided by `ember-hook`
* **Removed** ignoring of `package-lock.json` file
* **Added** `package-lock.json` file
* **Updated** Travis CI scripts to allow non-exact node version

### CODE REVIEW

File: .travis/maybe-bump-version.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if [ "$TRAVIS_NODE_VERSION" != "6.9.1" ]
+if [ "$TRAVIS_NODE_VERSION" != "8.1.2" ]
```
Comment (notmessenger) [inline .travis/maybe-bump-version.sh]: Should we change this to `if [[ ! "${PUBLISH_NODE_VERSION:-8.1.2}" =~ ^$TRAVIS_NODE_VERSION ]]`, introduced in https://github.com/ciena-blueplanet/pr-bumper/pull/136?

File: .travis/maybe-check-coverage.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if [ "$TRAVIS_NODE_VERSION" != "6.9.1" ]
+if [ "$TRAVIS_NODE_VERSION" != "8.1.2" ]
```
Comment (notmessenger) [inline .travis/maybe-check-coverage.sh]: Should we change this to `if [[ ! "${PUBLISH_NODE_VERSION:-8.1.2}" =~ ^$TRAVIS_NODE_VERSION ]]`, introduced in https://github.com/ciena-blueplanet/pr-bumper/pull/136?

File: .travis/maybe-publish-coverage.sh [hunk 3]
```diff
@@ -8,7 +8,7 @@ then
   exit 0
 fi
 
-if [ "$TRAVIS_NODE_VERSION" != "6.9.1" ]
+if [ "$TRAVIS_NODE_VERSION" != "8.1.2" ]
```
Comment (notmessenger) [inline .travis/maybe-publish-coverage.sh]: Should we change this to `if [[ ! "${PUBLISH_NODE_VERSION:-8.1.2}" =~ ^$TRAVIS_NODE_VERSION ]]`, introduced in https://github.com/ciena-blueplanet/pr-bumper/pull/136?

File: addon/helpers/array.js [hunk 4]
```diff
@@ -3,13 +3,12 @@
  * Temporary fix until the following bug is resolved: https://github.com/emberjs/ember.js/issues/14264
  */
 import Ember from 'ember'
-const {A, Helper} = Ember
+const {A: emberA, Helper} = Ember
```
Comment (notmessenger) [inline addon/helpers/array.js]: I know we'll be changing this at some point in the future (2.16+) but `emberArray` feels better than `emberA`.

Comment (juwara0) [reply addon/helpers/array.js]: done

File: .travis/maybe-bump-version.sh (hunk 1 above)
Comment (juwara0) [reply .travis/maybe-bump-version.sh]: done

File: .travis/maybe-check-coverage.sh (hunk 2 above)
Comment (juwara0) [reply .travis/maybe-check-coverage.sh]: done

File: .travis/maybe-publish-coverage.sh (hunk 3 above)
Comment (juwara0) [reply .travis/maybe-publish-coverage.sh]: done

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/546/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**